Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_baking.py

```diff
@@ -0,0 +1,1202 @@
+    bakedTextureSize: bpy.props.IntVectorProperty(name="Baked texture size", size=2,
+                                                  default=(512, 512))
+    bake_selection: bpy.props.EnumProperty(name="Objects to bake",
+                                           items=(('ALL', 'All',
+                                                   'Bake all objects in the scene (including hidden)'),
+                                                  ('SELECTED', 'Selected',
+                                                   'Bake only the selected object')),
+                                           default='ALL')
+    bake_channel_settings: bpy.props.CollectionProperty(type=MESHSYNC_BakeChannelSetting)
+    bake_all_channels: bpy.props.BoolProperty(name="All", description="Toggle all",
+                                              get=msb_bakeAllGet,
+                                              set=msb_bakeAllSet)
+    generate_uvs: bpy.props.EnumProperty(name="Generate UVs",
+                                         description="Whether to auto-generate UVs",
+                                         items=(('OFF', 'Off',
+                                                 'Bake all objects in the scene (including hidden)'),
+                                                ('IF_NEEDED', 'If needed',
+                                                 'Automatically UV unwraps objects if there are no UVs or existing UVs are not in the 0..1 range. WARNING: This will delete existing UVs on the object!'),
+                                                ('ALWAYS', 'Always',
+                                                 'Always automatically UV unwraps objects. WARNING: This will delete existing UVs on the object!')),
+                                         default='OFF')
+    apply_modifiers: bpy.props.BoolProperty(name="Apply modifiers",
+                                            description="In order to bake and get correct UVs, all modifiers need to be applied",
+                                            default=True)
+    run_modal: bpy.props.BoolProperty(name="Run Modal",
+                                            description="If this is enabled blender stays more interactive but baking is slower.",
+                                            default=False)
+    bake_progress: bpy.props.FloatProperty(
+        name="Progress",
+        subtype="PERCENTAGE",
+        min=0,
+        max=100,
+        precision=0)
+    bake_message: bpy.props.StringProperty()
+    bake_maps_remaining: bpy.props.StringProperty(name="Maps baked")
+    bake_time_remaining: bpy.props.StringProperty(name="Estimated time left")
+
+
+def msb_bakeAllChanged(self, context):
+    bakeAll = context.scene.meshsync_bake_settings.bake_all_channels
+    for channelSetting in context.scene.meshsync_bake_settings.bake_channel_settings:
+        channelSetting.bakeChannelEnabled = bakeAll
+
+
+@persistent
+def msb_setBakingDefaults(dummy):
```

Comment:
why do we need the dummy argument?

Reply:
This method is called from blender's load_post event, which requires the method to have 1 parameter. We don't use it.